### PR TITLE
feature: CrudTable

### LINKED ISSUE #605: CrudTable
We are using the EpicTable a lot for our entity listings and it would be a lot easier if there is a component that simplifies the EpicTable for that purpose.

### PR DESCRIPTION
**Pull request by gidomanders**
We are using the EpicTable a lot for our entity listings and it would be
a lot easier if there is a component that simplifies the EpicTable for
that purpose.

Added CrudTable component which wraps the EpicTable to make it easier to
display a table with create/edit/detail view and pagination.
Added CrudHeader component to simplify the display of a header with
sort and search functionality.

Closes #605

### CODE REVIEW

File: src/table/CrudTable/CrudTable.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,693 @@
...
+            }}
+          </EpicDetailRow>
+          <EpicDetailRow left={columns.firstName} active={creating}>
+            {() => (
+              <EpicDetail onClose={() => setCreating(false)}>
+                <PersonForm />
+              </EpicDetail>
+            )}
+          </EpicDetailRow>
+        </CrudTable>
```
Comment (TimQuist) [inline src/table/CrudTable/CrudTable.stories.tsx:644]: The example is very detailed which is good for viewing the component in the storybook, but I find the code example that you get pretty overwhelming.  Maybe it's an idea to also create a bare-bones version so people could view that and understand the 'fundamentals' of this CrudTable?

**Review by TimQuist — COMMENTED** [review]
Tested locally (in storybook). Seems to work just fine besides the minor 'bug' in the example

File: src/table/CrudTable/CrudTable.stories.tsx (hunk 1 above)
Comment (gidomanders) [reply src/table/CrudTable/CrudTable.stories.tsx:731]: @TimQuist I've added a "simple example" story. It's a very much thinned-out version of the full example where you don't have to scroll through that much code.

**Review by TimQuist — APPROVED** [review]

File: src/table/CrudTable/CrudTable.stories.tsx (hunk 1 above)
Comment (TimQuist) [reply src/table/CrudTable/CrudTable.stories.tsx:731]: Tested locally and looks good

**Pull request merged**